// react-contextmenu/index.js
(function () {
  'use strict';

  window.react_contextmenu = window.react_contextmenu || {};
  const exports = window.react_contextmenu;

  Object.defineProperty(exports, "__esModule", {
    value: true
  });

  var _ContextMenu = window.react_contextmenu.ContextMenu;

  Object.defineProperty(exports, 'ContextMenu', {
    enumerable: true,
    get: function get() {
      return _interopRequireDefault(_ContextMenu).default;
    }
  });

  var _ContextMenuTrigger = window.react_contextmenu.ContextMenuTrigger;

  Object.defineProperty(exports, 'ContextMenuTrigger', {
    enumerable: true,
    get: function get() {
      return _interopRequireDefault(_ContextMenuTrigger).default;
    }
  });

  var _MenuItem = window.react_contextmenu.MenuItem;

  Object.defineProperty(exports, 'MenuItem', {
    enumerable: true,
    get: function get() {
      return _interopRequireDefault(_MenuItem).default;
    }
  });

  var _SubMenu = window.react_contextmenu.SubMenu;

  Object.defineProperty(exports, 'SubMenu', {
    enumerable: true,
    get: function get() {
      return _interopRequireDefault(_SubMenu).default;
    }
  });

  var _connectMenu = window.react_contextmenu.connectMenu;

  Object.defineProperty(exports, 'connectMenu', {
    enumerable: true,
    get: function get() {
      return _interopRequireDefault(_connectMenu).default;
    }
  });

  var _actions = window.react_contextmenu.actions;

  Object.defineProperty(exports, 'hideMenu', {
    enumerable: true,
    get: function get() {
      return _actions.hideMenu;
    }
  });
  Object.defineProperty(exports, 'showMenu', {
    enumerable: true,
    get: function get() {
      return _actions.showMenu;
    }
  });

  function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

})();
